src/GNUmakefile

```diff
@@ -1,66 +1,43 @@
-CFLAGS := -g -W -Wall -Wno-unused-parameter -I. -DGMP -O2 -fno-omit-frame-pointer
-CXXFLAGS := $(CFLAGS) -std=c++11
-
-ifneq (,$(filter $(shell uname -n),elaines-MacBook-Pro.local Nicholass-Air-5.wireless.yale.internal))
-    LIBS := -L/usr/local/lib
-    CFLAGS += -I/usr/local/include
-else
-    LIBS :=
+CFLAGS := -g -W -Wall -Wno-unused-parameter -I. -O2 -fno-omit-frame-pointer
+LIBS := 
+ifndef NGMP
+    CFLAGS += -I/usr/local/include -DGMP
+    LIBS += -L/usr/local/lib -lgmp
 endif
 
-LIBS += -L/usr/local/lib -lgmpxx -lgmp
+CXXFLAGS := $(CFLAGS)
 
-ifneq ($(shell uname -n),beepboop)
-    LDFLAGS :=
-    CFLAGS += -fno-omit-frame-pointer
-else
-    LDFLAGS :=
-endif
-
-ifneq ($(shell uname),Darwin)
-    LIBS += #-lbsd
-endif
-
-CC = gcc
```

Comment:
I think we still want to use gcc instead of g++ to compile rulelib.c

Reply:
This would require creating a new pair of files to house my implementations of getline, strsep, and strdup (since they are currently used in rulelib.cpp, but declared in a c++ header file utils.hh). There might still be problems though, since I remember I moved rulelib.c to rulelib.cpp to make the build succeed on Windows.